From af2b65e8b95ef23d0f113654fe6969f4c26e1591 Mon Sep 17 00:00:00 2001
From: Michael Matz <matz@suse.com>
Date: Tue, 20 Sep 2016 11:50:37 +0100
Subject: Accept empty modules

For the Xen platform the all_video.mod module is empty.
With old binutils the .symtab section remained (containing
only section symbols), so the check didn't trigger, but starting
with binutils 2.27 not even a .symtab remains.  As there are
also no relocations that's no problem (and that is checked
independently).

Origin: vendor, https://savannah.gnu.org/file/grub2-accept-empty-module.patch?file_id=38435
Bug: https://savannah.gnu.org/bugs/?49012
Last-Update: 2016-09-20

Patch-Name: accept-empty-module.patch
---
 util/grub-module-verifierXX.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util/grub-module-verifierXX.c b/util/grub-module-verifierXX.c
index 9c04caa63..f90894006 100644
--- a/util/grub-module-verifierXX.c
+++ b/util/grub-module-verifierXX.c
@@ -199,7 +199,8 @@ check_symbols (const struct grub_module_verifier_arch *arch, Elf_Ehdr *e)
       Elf_Shdr *s = find_section (arch, e, ".moddeps");
 
       if (!s)
-	grub_util_error ("no symbol table and no .moddeps section");
+	/*grub_util_error ("no symbol table and no .moddeps section");*/
+	return; /* An empty module happens for all_video.module for Xen */
 
       if (!s->sh_size)
 	grub_util_error ("no symbol table and empty .moddeps section");
